discord_bots/cogs/admin.py

```diff
@@ -0,0 +1,692 @@
+                )
+                session.commit()
+            else:
+                if player.is_admin:
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description=f"{escape_markdown(player.name)} is already an admin",
+                            colour=Colour.red(),
+                        )
+                    )
+                else:
+                    player.is_admin = True
+                    session.commit()
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description=f"{escape_markdown(player.name)} added to admins",
+                            colour=Colour.green(),
+                        )
+                    )
+
+    @group.command(name="addrole", description="Add an admin role")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(role="Role to be made admin")
+    async def addadminrole(self, interaction: Interaction, role: Role):
+        if interaction.guild:
+            if not role in interaction.guild.roles:
+                await interaction.response.send_message(
+                    embed=Embed(
+                        description=f"Could not find role: {role.name}",
+                        colour=Colour.blue(),
+                    ),
+                    ephemeral=True,
+                )
+                return
+
+            session: SQLAlchemySession
+            with Session() as session:
+                admin_role: AdminRole | None = (
+                    session.query(AdminRole)
+                    .filter(AdminRole.role_id == role.id)
+                    .first()
+                )
+                if admin_role:
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description="Role is already an admin", colour=Colour.blue()
+                        ),
+                        ephemeral=True,
+                    )
+                else:
+                    session.add(AdminRole(role.id))
+                    session.commit()
+                    await interaction.response.send_message(
+                        embed=Embed(
+                            description=f"Added admin role: {role.name}",
+                            colour=Colour.green(),
+                        )
+                    )
+
+    @group.command(name="ban", description="Bans player from queues")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(member="Player to be banned")
+    async def ban(self, interaction: Interaction, member: Member):
+        """TODO: remove player from queues"""
+        session: SQLAlchemySession
+        with Session() as session:
+            players = session.query(Player).filter(Player.id == member.id).all()
```

Comment:
`.first()` can definitely be used here instead to simplify the logic, but since it was like this before you can just add a TODO. Up to you if you want to refactor it or not, because the logic that follows assumes the correct player is `players[0]`

Reply:
Replaced `players` with `player: Player | None`, and modified function logic to understand the singular nature of `.first()` as opposed to `.all()`